--- ./entity-name.js.orig	2018-10-22 22:01:03.335545400 +0300
+++ ./entity-name.js	2018-10-22 22:01:18.035584300 +0300
@@ -107,7 +107,7 @@
      * @deprecated use {@link BemEntityName#mod.name}
      */
     get modName() {
-        deprecate(this, 'modName', 'mod.name');
+        // deprecate(this, 'modName', 'mod.name');
 
         return this.mod && this.mod.name;
     }
@@ -121,7 +121,7 @@
      * @deprecated use {@link BemEntityName#mod.val}
      */
     get modVal() {
-        deprecate(this, 'modVal', 'mod.val');
+        // deprecate(this, 'modVal', 'mod.val');
 
         return this.mod && this.mod.val;
     }
